Update migration base.

nedSaf opened the pull request.

nedSaf commented:
@AronNovak @amitaibu @bitamar 
This is an update to the custom migration module.

I know this migration only has one handler for nodes and one for users but this is being copied into new projects and I would like to have all the common code and settings in the base and not in the handlers themselves. 

The advantage is that unfamiliar developers won't have to copy this whole code to each new handler and familiar ones won't have to create their own new base.

Please review, whoever has a little bit of time, thanks :)

**_Note:_** I would like to add much more comments in the base to explain what each thing does but I'm not sure if it's just unnecessary.

nedSaf commented:
No errors:
![default___hedley](https://user-images.githubusercontent.com/7369740/43999890-8c8d60de-9e1e-11e8-944e-9d0e6cb7534c.png)

AronNovak commented on server/hedley/modules/custom/hedley_migrate/handlers/HedleyMigrateBase.php:
```diff
@@ -10,6 +10,87 @@
+    $destination_class = $destination_classes[$this->entityType];
+    $this->destination = $this->entityType == 'user' ? new $destination_class() : new $destination_class($this->bundle);
+
+    $key_field = $this->bundle == 'user' ? 'name' : 'id';
```
If this is the case, why do we introduce it at all in ` server/hedley/modules/custom/hedley_migrate/csv/user.csv` ?

AronNovak replied:
or we could get rid of it and entirely rely on `id` for the sake of consistency

AronNovak commented on server/hedley/modules/custom/hedley_migrate/csv/item.csv:
```diff
@@ -1,4 +1,4 @@
-Kitchen,kitchen.jpg,Secret note
-West Garden,west-garden.jpg,
-<script>alert('XSS');</script>XSS Test,kitchen.jpg,
+id,title,field_image,field_private_note
```
I guess this might be the right time to add some example how to to cross-CSV references, so:
`id,title,field_image,field_private_note,author_name`

or 
`id,title,field_image,field_private_note,author_id`

Based on my other comment I made

AronNovak commented:
@nedSaf I like it generally, I added a few minor comments.

nedSaf commented:
Migration:
![default___hedley](https://user-images.githubusercontent.com/7369740/44192728-3b9aab00-a139-11e8-9728-1bafbf5acdd8.png)

Content:
![content___hedley](https://user-images.githubusercontent.com/7369740/44192749-50773e80-a139-11e8-99e7-bf74da550bc8.png)

nedSaf commented:
## Update
- Fixed the comments.
- Upgraded the migrate module's version.
- Added some more docs for vars and logic in the base migration class.

nedSaf commented:
@AronNovak Ready for a second review :)
